### PR TITLE
Gracefully handle forked attestation

### PR DESCRIPTION
**Pull request by terencechain**
We currently don't handle forked attestation well. If there's a forked attestation which the state can't process (eg. don't have validator index), the attestation will return an error and we lost it.

This PR saves the forked attestation and re-process it upon every successful state transition. It also deletes the forked attestation when it's one epoch old

This is controlled via feature flag `enable-forked-attestation-processing`

### CODE REVIEW

File: beacon-chain/attestation/service.go [hunk 1]
```diff
@@ -246,59 +248,29 @@ func (a *Service) updateAttestation(ctx context.Context, headRoot [32]byte, beac
 	attestation *pb.Attestation) error {
 	totalAttestationSeen.Inc()
 
-	slot := attestation.Data.Slot
```
Comment (terencechain) [inline beacon-chain/attestation/service.go]: refactored this to `committee` function

File: beacon-chain/attestation/service.go [hunk 2]
```diff
@@ -308,14 +280,6 @@ func (a *Service) updateAttestation(ctx context.Context, headRoot [32]byte, beac
 			continue
 		}
 
-		if i >= len(committee) {
```
Comment (terencechain) [inline beacon-chain/attestation/service.go]: refactored this to `canProcessBitfield` function

File: beacon-chain/attestation/service.go [hunk 3]
```diff
@@ -246,59 +249,29 @@ func (a *Service) updateAttestation(ctx context.Context, headRoot [32]byte, beac
...
-		if v.Shard == attestation.Data.Shard {
-			committee = v.Committee
-			break
+	bitfield := attestation.AggregationBitfield
+	if !a.canProcessBitfield(committee, bitfield, len(beaconState.ValidatorRegistry)) {
+		if featureconfig.FeatureConfig().EnableForkedAttestationProcessing {
+			a.forkedAttestations = append(a.forkedAttestations, attestation)
 		}
+
+		return fmt.Errorf("invalid state to process forked attestation at slot %d",
```
Comment (nisdas) [inline beacon-chain/attestation/service.go]: you should make log this as a debug , and then return it

Comment (terencechain) [reply beacon-chain/attestation/service.go]: got it, thanks for the feedback!

**Comment by terencechain** [comment]
we no longer need this since fork choice uses validators from last finalized state, and when there's a new finalized state > current finalized state the beacon node should just switch over there

**Pull request closed without merging**